### ISSUE TITLE
WATCHDOG: GPU 4 hangs in OpenCL call, exit ... need to restart ...

### ISSUE THREAD

**Issue by daryong** [nanopool/Claymore-Dual-Miner#140]
Hi,

I am trying to mining 12 gpu from ubuntu.
With 8 gpu, the window 10 has already been successfully mined.

However, when you try to mine in ubuntu, you will get an error such as GPU hangs randomly
Claymore wisely restarts the program at this time.
However, every time there is a decrease in gpu memory and eventually a DAG alloc error
It can no longer be mined.

In Windows, Claymore did not go down, so when I looked up the log, Windows sometimes restarted the program. But I found a difference with Linux.

< Claymore Restart Log - **Windows** >
17:59:47:926	f80	WATCHDOG: GPU 4 hangs in OpenCL call, exit
17:59:47:942	f80	watchdog - thread 10 (gpu5), hb time 297
17:59:47:957	f80	watchdog - thread 11 (gpu5), hb time 156
17:59:47:957	f80	watchdog - thread 12 (gpu6), hb time 296
17:59:47:973	f80	watchdog - thread 13 (gpu6), hb time 172
17:59:47:988	f80	watchdog - thread 14 (gpu7), hb time 219
17:59:48:004	f80	watchdog - thread 15 (gpu7), hb time 78
**17:59:48:254	f80	OC v7, Reset control for GPU 0, close miner right now if you want to use default control from Catalyst 
17:59:48:270	f80	OC v7, Reset control for GPU 1, close miner right now if you want to use default control from Catalyst 
17:59:48:270	f80	OC v7, Reset control for GPU 2, close miner right now if you want to use default control from Catalyst 
17:59:48:270	f80	OC v7, Reset control for GPU 3, close miner right now if you want to use default control from Catalyst 
17:59:48:442	f80	OC v7, Reset control for GPU 4, close miner right now if you want to use default control from Catalyst 
17:59:48:442	f80	OC v7, Reset control for GPU 5, close miner right now if you want to use default control from Catalyst 
17:59:48:457	f80	OC v7, Reset control for GPU 6, close miner right now if you want to use default control from Catalyst 
17:59:48:457	f80	OC v7, Reset control for GPU 7, close miner right now if you want to use default control from Catalyst** 
17:59:49:583	f80	Restarting OK, exit...

In other words, there seems to be **no gpu reset process on linux**. So, in linux, restarting seems to be abnormal.

At present, ubuntu installs only amdgpu-pro-17.40 and runs claymore.

Claymore's bug?
Is this happening because there are libraries that are not installed?

Thank you.

**Comment by humairarehan** [comment]
can anyone help on this

**Comment by aalbul** [comment]
Well, i didn't try this on ubuntu, but from my experience, this usually happens when you overclocked your cards to much. Try to reduce overclock / under-voltage a bit.
Another solution that can be helpfull - turn off claymore's watchdog. A friend of mine didn't want to reduce overclock so he disabled watchdog and that helped him :)

You can do that like this: -wd 0

**Comment by Cavien** [comment]
ETHOS 遇到同样的问题。

**Comment by exeex** [comment]
I am facing the same issue on ubuntu 16.04

after receiving any GPU hangs in OpenCL call

I can't not restart claymore normaly, even reboot my computer! 
(sudo reboot would stop while shutdown)

ETH - Total Speed: 308.682 Mh/s, Total Shares: 22379, Rejected: 0, Time: 72:28
ETH: GPU0 31.048 Mh/s, GPU1 30.105 Mh/s, GPU2 31.059 Mh/s, GPU3 31.068 Mh/s, GPU4 31.068 Mh/s, GPU5 31.054 Mh/s, GPU6 0.000 Mh/s, GPU7 0.000 Mh/s, GPU8 30.113 Mh/s, GPU9 31.047 Mh/s, GPU10 31.067 Mh/s, GPU11 31.052 Mh/s
WATCHDOG: GPU 6 hangs in OpenCL call, you need to restart miner :(
WATCHDOG: GPU 6 hangs in OpenCL call, you need to restart miner :(
WATCHDOG: GPU 7 hangs in OpenCL call, you need to restart miner :(
WATCHDOG: GPU 7 hangs in OpenCL call, you need to restart miner :(
WATCHDOG: GPU error, you need to restart miner :(

**Comment by qnarkill** [comment]
I suffer in that too, changing PEG1 to gen2 and PEG0 to gen3 from auto plus increasing pci latency seem to 64 helped a bit. Also i removed all the rubber protectives from the io-ports.

**Comment by romanap25** [comment]
Not sure if this will help anyone but after almost 1 month Of having the same issue, my problem turned out to be with PSU. On my previous rigs I used Corsair RM1200 PSU's but on the new one I have HX1200 which has a switch on the back between single and multiple. It was set on multiple and after an hour or sometimes even 2 minutes a random card would fail. Since I see itched it to single my rig has now been working for over 3 days without any issues what so ever. According to Corsair if the switch is in 'multiple' position each connector has Over Current Protection so if there is current surge in your outlet it will limit the power hence why a GPU would not get enough power and stop working. Try it and let me know if it solves the problem.

**Comment by qnarkill** [comment]
I actually had 2 broken cards under warranty, removing those helped.

**Comment by clovanzo** [comment]
@romanap25 : You just save such pain in ass, i face same issues now my rig just run smoothly, thank you very much

**Comment by sandeepkumare** [comment]
hello my name is saneep   mining setup Main Agar kuch bhi dikkat hai toh main soul kar sakta Hu free mai   CLAYMORE FREEZE ETHEREUM MINING "WATCHDOG: GPU 3 hangs in OpenCL call, exit - HOW TO FIX IT  < Claymore Restart Log - Windows > after receiving any GPU hangs in OpenCL call   my phone no. 7427893256

**Comment by sandeepkumare** [comment]
> Pick your reaction
hello my name is saneep mining setup Main Agar kuch bhi dikkat hai toh main soul kar sakta Hu free mai 
my phone no. 7427893256

**Comment by kishan143** [comment]
I have the same problem, reinstalled all driver, etc done, still problem persist...

**Comment by kishan143** [comment]
@sandeepkumare

Your number is not working ....
Kindly privide alternative number

**Comment by sandeepkumare** [comment]
ok bhai  my new phone no- 6375970070

**Comment by tdani30** [comment]
Hello,

----- watchdog gpu 0 hangs in opencl call exit ----------------

I'm encountering one issue, while mining from "https://ethermine.org". I'm using ethdcrminer64.exe for mining with Nvidia card, but after few minutes it got stuck and never come back until I quit the miner and start again. Please guide me on that. I also attach one file, for reference please find the attachment.
Thanks
![miniing](https://user-images.githubusercontent.com/19394776/43194038-82f60e96-9012-11e8-80d0-f051e6d77c18.PNG)
